drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -1808,6 +1808,8 @@ static int vc4_hdmi_encoder_atomic_check(struct drm_encoder *encoder,
 	struct vc4_hdmi_connector_state *vc4_state = conn_state_to_vc4_hdmi_conn_state(conn_state);
 	struct drm_display_mode *mode = &crtc_state->adjusted_mode;
 	struct vc4_hdmi *vc4_hdmi = encoder_to_vc4_hdmi(encoder);
+	unsigned int output_bpc = vc4_state->output_bpc;
+	enum vc4_hdmi_output_format output_format = vc4_state->output_format;
```

Comment:
Even though it probably works because nothing modified it so far, we should directly compare with the old connector state here. You can retrieve it using 
```
struct drm_connector_state *old_conn_state = drm_atomic_get_old_connector_state(conn_state->state, connector);
struct vc4_hdmi_connector_state *old_vc4_state = conn_state_to_vc4_hdmi_conn_state(old_conn_state);
```